.github/workflows/valid-files.yml

```
@@ -0,0 +1,55 @@
+name: Check if files are valid
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  max-file-size:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if any file is above 1 MB
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
+
+          for FILE in $FILES; do
+            if [ "$(stat --printf="%s" "$FILE")" -gt 100000 ]; then
+              echo "$FILE is larger than 10 KB"
+              echo "Please ping @iHiD if you are sure that this file is required."
+              exit 1
+            fi
+          done
+
+  valid-filenames:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if filenames contain illegal characters or words
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
+
+          for FILE in $FILES; do
+            if [[ "$FILE" =~ ^[a-zA-Z0-9._-]+$ ]]; then
+              echo "$FILE contains illegal characters."
+              echo "Filenames in this repo may only include the characters [a-zA-Z0-9._-] for security reasons."
+              echo "See https://github.com/exercism/v3/issues/1465#issuecomment-629873323 for more information."
+            elif [[ "$FILE" =~ ^(AUX|NUL|PRN|CON|LPT[0-9]|COM[0-9]) ]]; then
```

I think this needs a `$` at the end, as `AUX` _is_ invalid but `AUX1` e.g. isn't.

```suggestion
            elif [[ "$FILE" =~ ^(AUX|NUL|PRN|CON|LPT[0-9]|COM[0-9])$ ]]; then
```